public/views/jQuery.js

```diff
@@ -0,0 +1,141 @@
+$(document).ready(function() {
+
+  scorecard = new BowlingScorecard();
+
+  var outputCounter = 0;
+  var numberChecker = 0;
+  printTotal();
+
+  $("#button0").on('click', function() {
```

Comment:
I think there are several improvements we can make here. 

First, rather than assigning an anonymous function as a click handler, it's usually better to use a named function, e.g. 

```
function handleButtonClick() {
   ..code...
}

$(document).ready(function() {
   $("#button0").on('click', handleButtonClick);
});
``` 

The advantages of this are that the handleButtonClick function is now testable independently of the JQuery that it is handling - we've separated the concern of reacting to an event from how we react to that event.

Second, there's a lot of repetition here. Rather than assigning the click handlers individually, we could generalise, e.g.

``` 
   $(document).ready(function() {
      var scoreButtons = $('.scoreButton');
      for (var index = 0; index < scoreButtons.length; index++ {
         scoreButtons[0].on('click', handleButtonClick);
      }
   });
``` 

But how do we assign different values to each of the buttons in this case? Well, there's nothing to stop us passing information into handleButtonClick:

``` 
   function handleButtonClick(buttonElement, rollValue) {
      $(buttonElement).html(rollValue);
      printButton(rollValue);
      etc...
   }

   $(document).ready(function() {
      var scoreButtons = $('.scoreButton');
      for (var index = 0; index < scoreButtons.length; index++ {
         scoreButtons[0].on('click', function() {
            return handleButtonClick(scoreButtons[0], scorecard.roll(index));
         });
      }
   });

```